Cargo.toml

```diff
@@ -9,7 +9,7 @@ readme = "README.md"
 exclude = [".gitignore", ".gitmodules", "examples", "lightgbm-sys"]
 
 [dependencies]
-lightgbm-sys = "0.2.0"
+lightgbm-sys = { path = "lightgbm-sys", version = "0.2.0" }
```

Comment:
[ASK] Is there a reason to use the path?

Reply:
This change makes cargo use modified local lightgbm-sys.

I edited the build script and tried to build examples. But, cargo uses the lightgbm-sys published on crates.io instead of modified one.

However, I am not familiar with cargo, so maybe there is a right way to solve it. I would appreciate it if you could tell me.


Another.
I'm writing rust code with vscode + rust-analyzer extension.

When opening project root directory with vscode.
Without the path, rust-analyzer didn't read the source code of lightgbm-sys in the subdirectory.
With path, rust-analyzer read lightgbm-sys crate and works.

Reply:
@paq Thanks answer. You're saying is right:)

According to the following, you can separate the create.io and local behavior by writing `version=`.
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations

I didn't know the details either. Thank you.

Reply:
This seems to be causing the CI to field. It's probably because I'm trying to build LightGBM. I'll check.